test/functional/tests/volumes/test_cache1_core4_lvm4.py

```diff
@@ -0,0 +1,121 @@
+          1 cache, 4 cores, 4 lvms.
+        pass_criteria:
+          - CAS devices created successfully.
+          - LVMs created successfully.
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Create CAS device."):
+
+        cache_dev, core_devices = prepare_devices()
+
+        cache = casadm.start_cache(cache_dev[0], force=True)
+        cores = []
+        for core_dev in core_devices:
+            cores.append(cache.add_core(core_dev))
+
+    with TestRun.step("Create PV, VG and LV on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        config = LvmConfiguration(lvm_filters)
+        lvms = []
+        for core in cores:
+            lvm = Lvm.create(Size(5, Unit.GiB), core, lvm_filters=config)
+            lvms.append(lvm)
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Flush buffers"):
+        for lvm in lvms:
+            TestRun.executor.run(f"hdparm -f {lvm.path}")
+
+    with TestRun.step("Create init config from running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_before_reboot = fs_utils.read_file(opencas_conf_path)
+
+    with TestRun.step("Reboot system."):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Validate running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_after_reboot = fs_utils.read_file(opencas_conf_path)
+
+        if config_after_reboot == config_before_reboot:
+            TestRun.LOGGER.info(f"Configuration is as expected")
+        else:
+            TestRun.LOGGER.error(f"Configuration changed after reboot.")
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Remove LVMs."):
+        Lvm.remove_all()
+
+
+def prepare_devices():
+    cache_device = TestRun.disks['cache']
+    core_devices = [TestRun.disks['core1'],
+                    TestRun.disks['core2']]
+
+    cache_device.create_partitions([Size(10, Unit.GibiByte)])
+
+    core_partitions = []
+    for core_dev in core_devices:
+        core_dev.create_partitions([Size(8, Unit.GibiByte)] * 2)
+        core_partitions.append(core_dev.partitions[0])
+        core_partitions.append(core_dev.partitions[1])
+    sync()
+
+    return cache_device.partitions, core_partitions
+
+
+def run_fio_on_lvm(volumes):
+    fio_run = Fio().create_command()
+    fio_run.read_write(ReadWrite.randrw)
+    fio_run.io_engine(IoEngine.sync)
+    fio_run.io_depth(1)
+    fio_run.time_based()
+    fio_run.run_time(datetime.timedelta(seconds=180))
+    fio_run.do_verify()
+    fio_run.verify(VerifyMethod.md5)
+    fio_run.block_size(Size(1, Unit.Blocks4096))
```

Comment:
this is not something you have to change but I wanted to inform you that you can create fio command without calling fio_run in each line, you can do this by calling all fio parameters one by one after the dot